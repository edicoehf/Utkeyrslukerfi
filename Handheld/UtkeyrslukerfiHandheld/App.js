import 'react-native-gesture-handler'
import React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import ScanScreen from './src/views/ScanScreen'
import ListScreen from './src/views/ListScreen'
import SearchScreen from './src/views/SearchScreen'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
import MaterialCommunityIcon from 'react-native-vector-icons/MaterialCommunityIcons'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
import { createStackNavigator } from '@react-navigation/stack'

const Tab = createBottomTabNavigator()
const SearchStack = createStackNavigator()

const SearchStackScreen = () => {
  return (
    <SearchStack.Navigator
      initialRouteName='Search'
      screenOptions={{
        headerShown: true,
        title: 'Search',
        headerStyle: {
            backgroundColor: '#fafafa',
        },
        headerTitleStyle: {
            fontWeight: 'bold',
        }
      }}
    >
      <SearchStack.Screen name='Search' component={SearchScreen} />
    </SearchStack.Navigator>
  )
}

const App = () => {
  return (
    <NavigationContainer>
      <Tab.Navigator initialRouteName='Scan'>
        <Tab.Screen 
          name='List' 
          component={ListScreen} 
          options={{
            showIcon: true,
            tabBarLabel: 'List',
            tabBarIcon: () => <MaterialCommunityIcon name='format-list-bulleted' style={{width:26-32}} color='#333' size={24} />
          }}
        />
        <Tab.Screen 
          name='Scan'
          component={ScanScreen}
          options={{
            showIcon: true,
            tabBarLabel: 'Scan',
            tabBarIcon: () => <MaterialCommunityIcon name='barcode-scan' style={{width:26-32}} color='#333' size={24} />
          }}
        />
        <Tab.Screen 
          name='Search' 
          component={SearchStackScreen} 
          options={{
            showIcon: true,
            tabBarLabel: 'Search',
            tabBarIcon: () => <MaterialIcon name='search' style={{width:26-32}} color='#333' size={24} />
          }}
        />
      </Tab.Navigator>  
    </NavigationContainer>
  )
}

export default App
