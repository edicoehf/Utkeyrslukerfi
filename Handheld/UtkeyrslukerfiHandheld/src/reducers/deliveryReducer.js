import { GET_DELIVERIES } from '../constants'

const deliveryReducer = (state = [], action) => {
  switch (action.type) {
    case GET_DELIVERIES: return action.payload
    default: return state
  }
}

export default deliveryReducer
