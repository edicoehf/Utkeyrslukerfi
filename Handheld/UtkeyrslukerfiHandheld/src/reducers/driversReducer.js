import { GET_DRIVERS } from '../constants'

const driversReducer = (state = [], action) => {
  switch (action.type) {
    case GET_DRIVERS: return action.payload
    default: return state
  }
}

export default driversReducer
