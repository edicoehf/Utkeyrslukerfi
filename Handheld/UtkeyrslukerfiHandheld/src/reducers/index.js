import { combineReducers } from 'redux'
import statusCode from './statusCodeReducer'
import login from './loginReducer'
import drivers from './driversReducer'
import deliveries from './deliveryReducer'

export default combineReducers({
  statusCode,
  login,
  drivers,
  deliveries
})
