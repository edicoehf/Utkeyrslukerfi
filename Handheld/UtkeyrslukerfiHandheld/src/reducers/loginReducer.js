import { SET_LOGIN, GET_LOGIN, CLEAR_LOGIN } from '../constants'

const loginReducer = (state = { token: '', driver: '' }, action) => {
  switch (action.type) {
    case SET_LOGIN: return action.payload
    case GET_LOGIN: return action.payload
    case CLEAR_LOGIN: return action.payload
    default: return state
  }
}

export default loginReducer
