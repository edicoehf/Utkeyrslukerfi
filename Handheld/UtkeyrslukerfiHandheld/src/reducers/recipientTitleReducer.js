import { SET_RECIPIENT_TITLE, GET_RECIPIENT_TITLE } from '../constants'

const recipientTitleReducer = (state = '', action) => {
  switch (action.type) {
    case SET_RECIPIENT_TITLE: return action.payload
    case GET_RECIPIENT_TITLE: return action.payload
    default: return state
  }
}

export default recipientTitleReducer
