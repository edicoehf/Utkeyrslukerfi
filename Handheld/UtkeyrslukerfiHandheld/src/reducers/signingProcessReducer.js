import configData from '../constants/config.json'
import { SET_STEP } from '../constants'

const signingProcessReducer = (state = { process: configData.SIGNING_PROCESS, step: 0 }, action) => {
  switch (action.type) {
    case SET_STEP: return { step: action.payload, ...state }
    default: return state
  }
}

export default signingProcessReducer
