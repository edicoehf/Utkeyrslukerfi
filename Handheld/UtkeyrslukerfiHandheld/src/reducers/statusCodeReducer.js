const statusCode = { 1: 'Í ferli', 2: 'Á leiðinni', 3: 'Móttekið' } // TODO: make status codes configurable

const statusCodeReducer = (state = statusCode, action) => {
  switch (action.type) {
    default: return state
  }
}

export default statusCodeReducer
