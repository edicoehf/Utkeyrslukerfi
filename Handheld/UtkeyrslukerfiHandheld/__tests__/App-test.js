/**
 * @format
 */

import 'react-native'
import React from 'react'
import App from '../App'
import { createStore, applyMiddleware } from 'redux'
import reducers from '../src/reducers'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
import AppContainer from '../src/routes'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

it('App renders correctly', () => {
  const app = renderer.create(
  <Provider store={createStore(reducers, applyMiddleware(thunk))}>
    <App />
  </Provider>
  ).toJSON()
  expect(app).toMatchSnapshot()
})

it('App container renders correctly', () => {
  const list = renderer.create(
    <Provider store={createStore(reducers, applyMiddleware(thunk))}>
        <AppContainer />
    </Provider>
  ).toJSON()
  expect(list).toMatchSnapshot()

})

// it('checks if Async Storage is used', async () => {
//   await asyncOperationOnAsyncStorage();

//   expect(AsyncStorage.getItem).toBeCalledWith('token');
// })
