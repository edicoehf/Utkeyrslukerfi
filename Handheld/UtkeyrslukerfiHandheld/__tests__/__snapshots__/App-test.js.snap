// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 15,
        "marginHorizontal": 5,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingLeft": 1,
        }
      }
    >
      Innskráning
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "zIndex": 5000,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FAFAFA",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderColor": "#dfdfdf",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {},
              Object {
                "flex": 1,
                "opacity": 1,
              },
              Object {
                "marginLeft": 5,
                "marginRight": 0,
              },
              [Function],
              Object {},
              false,
              Object {},
            ]
          }
        >
          Velja starfsmann...
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 0,
                "borderTopLeftRadius": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingVertical": 8,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 15,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FAFAFA",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "#dfdfdf",
              "borderTopLeftRadius": 0,
              "borderTopRightRadius": 0,
              "borderWidth": 1,
              "justifyContent": "center",
              "maxHeight": 120,
              "minHeight": 40,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
              "position": "absolute",
              "width": "100%",
            },
            Object {
              "borderWidth": 0,
              "display": "none",
              "position": "relative",
            },
            Object {
              "maxHeight": 150,
              "top": 0,
              "zIndex": 5000,
            },
          ]
        }
      >
        <RCTScrollView
          nestedScrollEnabled={true}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginBottom": 15,
                  "marginVertical": 10,
                }
              }
            >
              <Text>
                Not Found
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#81ABE4",
          "borderRadius": 5,
          "height": 93.38000000000001,
          "justifyContent": "center",
          "margin": "2%",
          "opacity": 1,
          "width": 187.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Skrá inn
      </Text>
    </View>
  </View>
</View>
`;
